### ISSUE TITLE
Can't [B]utcher a book

### ISSUE THREAD

**Issue by sparr** [CleverRaven/Cataclysm-DDA#11861]
I'm standing on a book and when I press [B] it says there are no corpses to butcher. I can pick the book up and disassemble it with [(], though. This is one example of what I think is a broader problem that mostly applies to less common disassembly items.
##

**Comment by BevapDin** [comment]
Disassembling a book is a special case. See the end of `player::disassemble` in crafting.cpp it's kind of a hack, but it completely bypasses the crafting system.

**Comment by drbig** [comment]
I think there is a general problem with the dichotomy between butchering and disassembly.

If you ask me (and this might be slightly contrived, sorry):
 - Butchering is the name for the *disassembly* action on corpses
 - Butchering of *non-corpses* is "cut it up, fast!" or "smash it up, fast!"
 - Therefore there is big difference between butchering and disassembly for *non-corpses*: You can butcher all the clothes on the ground to get some raw materials back; butchering *non-textile* items is usually a poor idea (you should get little usable stuff out of it)
 - Disassembly of both *textile* and *non-textile* items should only be possible on per-item basis, as you need to concentrate on the task to get as many usable 'parts' out of the item; e.g. butchering a radio is equivalent to stomping on it until it breaks open and taking whatever looks whole - disassembly of a radio is actually sitting down, taking a screwdriver and carefully taking it apart

Having said that I'd actually lean more towards "butchering is only an action on corpses" - but as we already have the butchering available on non-corpses... I think we can make it saner.

**Comment by narc0tiq** [comment]
> I think there is a general problem with the dichotomy between butchering and disassembly.

More like a trichotomy: butchering, disassembly, and salvage.

The history is such:
- First there was the [B]utcher action (currently activity `ACT_BUTCHER`). Worked (and still does) only on corpses, returned corpse stuff like chunks of meat and fat and such. Initially, it included the code to let you select among multiple corpses on the ground (i.e. "the butcher menu").
- Then robots started dropping corpse-like objects, which could not be butchered because that didn't make any sense (chunks of robotic meat?). Instead, they had disassembly recipes (unlearnable reversible recipes; disassembly itself is currently implemented as the activity `ACT_DISASSEMBLE`). You had to pick up the corpses and use the disassemble hotkey to get there. This being a little inconvenient, the [B]utcher menu was then extended to gain the ability to disassemble things on the ground, making robot corpses feel much like regular corpses, even if internally they used completely different code.
- Then, finally, someone took a close look at a feature of the `use_action: KNIFE`, namely the one that lets you apply the knife to an item and get back base materials (e.g. plastic chunks) based on what the item's made of (currently implemented as `iuse::cut_up`). They extended the system to calculate the materials better (in the process allowing items to have any number of materials, as opposed to just two) and named it "salvage". They then added this **third** activity to the already-extended [B]utcher-and-disassemble menu.

As such, the [B]utcher menu is actually [B]utcher-and-disassemble-and-salvage. The only intent to overloading the menu like this was to make it somewhat more convenient to trigger the two lesser actions, but the three actions I just mentioned are the distinctions the game currently understands.

To summarize:
- Butchering does _not_ trigger a disassembly of corpses. The butcher __menu__ includes disassembly and salvage entries.
- "Butchering of non-corpses" doesn't exist.
- There are completely different code paths involved in disassembly of items (reversing a crafting recipe; takes time, may consume batteries, etc.) and salvage (applying the knife "cut up" action, takes little time, is mostly hard-coded).

**Comment by illi-kun** [comment]
### This issue was closed as it appears inactive.

Reducing open issues to those which are (or will) be actively worked upon helps us focus our efforts. This issue has not been deleted - it still appears in searches and if it contains relevant information you are encouraged to continue to link to it.

### If this issue was a bug
It should be reopened if it can be reproduced in the current build. You can obtain the most recent copy [here](http://dev.narc.ro/cataclysm/jenkins-latest/). Please check there is not a more recent report of this bug before doing so. If no more recent report exists you should continue the discussion in this issue.

### If this was a feature request
If the consensus was that the idea was good you could consider submitting an implementation via a PR. If you want to comment further please do so here as opposed to opening a new issue. Before posting check nobody has already made the same point and consider whether your comments are likely to lead to an implementation. If you have doubts about either consider instead [voting](http://iv.cataclysmdda.com:8111/all) for the issue

### If you want to work on this issue
Then either assign it to yourself or if you are unable to do so claim it via adding a comment. Please don't assign others or make a general request for action.
